### DISCUSSION TITLE
Setting TZ Env Variables

### DISCUSSION THREAD

**Discussion by amcdnl** [vitest-dev/vitest#1718] [Q&A]
I have unit tests that rely on the `TZ` node variable ( https://stackoverflow.com/questions/8083410/how-can-i-set-the-default-timezone-in-node-js ). In Vitest, I do the following and everything works great:

```
    "test": "TZ=UTC vitest run",
```

However, when I use tools like the VSCode plugin which runs the tests differently, this is not getting picked up. I was wondering if there was a way to do this programmatically or via config so that other runners would pick it up.

**Comment by sheremet-va** [comment]
You can use `vitest.nodeEnv` option: https://marketplace.visualstudio.com/items?itemName=ZixuanChen.vitest-explorer

**Reply by sheremet-va** [reply]
You can also try and use `test.env` option in Vitest config file.

**Reply by amcdnl** [reply]
@sheremet-va - I tried doing the `test.env` approach and although validated it was actually get applied to the env, it did not work ( I wonder if that is a bug? ). Here you can see the `TZ` is set but the test that it would affect is failing.

<img width="497" alt="image" src="https://user-images.githubusercontent.com/227909/180873751-9763d126-b666-42f1-a0b2-c05d80445f98.png">

PA: I don't see a `env` option in the docs: https://vitest.dev/config/

**Comment by calvinf** [comment]
Add a globalSetup file:

```ts
// globalSetup.ts
export const setup = () => {
  // default all tests to UTC
  process.env.TZ = 'UTC';
};
```

Update your `vite.config.ts`:

```ts
test: {
  globalSetup: ['globalSetup.ts'],
  // ... other settings
}
```

**Reply by junqdu** [reply]
I just had `process.env.TZ = 'UTC';` in the file you have defined for `test.setupFiles` in `vite.config.ts`
